corehq/apps/userreports/tasks.py

```diff
@@ -288,6 +289,9 @@ def save_document(doc_ids):
     timer = TimingContext()
     with CriticalSection(lock_keys):
         indicators = AsyncIndicator.objects.filter(doc_id__in=doc_ids)
+        # proactive method: also grab related doc ids from AsyncIndicator and get their locks
+        # note: these only be rebuilt if they are also in the queue
+        # question: What happens if these child docs are being processed by another queue?
```

Comment:
practically if they are already in another queue we can just not process them, right? since they'll be handled there? is there an easy way in our locking framework to say "try and grab this lock and if you don't succeed just return immediately"?

Reply:
Yeah. that make sense. I think we can use `CriticalSection(blocking=False)` to do that sort of thing.